### DISCUSSION TITLE
not able to connect to docker socket in custom runner image

### DISCUSSION THREAD

**Discussion by shichengripple001** [actions/actions-runner-controller#1139] [Questions]
I am trying to build my own runner image, i have install docker into my image, my runner pod is running fine, I can even kubectl exec into my pod to get docker version info with no issue, using curl command
curl --cacert ca.crt --cert cert.crt --key key.key https://localhost:2376/version
{"Platform":{"Name":"Docker Engine - Community"},"Components":[{"Name":"Engine","Version":"20.10.12","Details":{"ApiVersion":"1.41","Arch":"amd64","BuildTime":"2021-12-13T11:46:12.000000000+00:00","Experimental":"false","GitCommit":"459d0df","GoVersion":"go1.16.12","KernelVersion":"5.4.172-90.336.amzn2.x86_64","MinAPIVersion":"1.12","Os":"linux"}},{"Name":"containerd","Version":"v1.4.12","Details":{"GitCommit":"7b11cfaabd73bb80907dd23182b9347b4245eb5d"}},{"Name":"runc","Version":"1.0.2","Details":{"GitCommit":"v1.0.2-0-g52b36a2d"}},{"Name":"docker-init","Version":"0.19.0","Details":{"GitCommit":"de40ad0"}}],"Version":"20.10.12","ApiVersion":"1.41","MinAPIVersion":"1.12","GitCommit":"459d0df","GoVersion":"go1.16.12","Os":"linux","Arch":"amd64","KernelVersion":"5.4.172-90.336.amzn2.x86_64","BuildTime":"2021-12-13T11:46:12.000000000+00:00"}

However when I trigger a github pipeline to run a dind job, i am getting:

  /usr/local/bin/docker version --format '{{.Server.APIVersion}}'
  Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

how is the runner calling the docker API? any way i can debug this issue?

**Comment by toast-gear** [comment]
ARC isn't a general purpose runner orchestrator, no idea
